### PR TITLE
[zkd] testInBox speedup

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose

This is a feature branch for #13650. This is a speedup for the testInBox and compareWithBox function.

### CODE REVIEW

File: arangod/Zkd/ZkdHelper.cpp [hunk 1]
```diff
@@ -264,37 +272,91 @@ auto zkd::compareWithBox(byte_string_view cur, byte_string_view min, byte_string
...
+        if (--finished_dims == 0) {
+          break;
+        }
+      }
+    }
+
+    dim += 1;
+    if (dim >= dimensions) {
+      dim = 0;
+    }
```
Comment (goedderz) [inline arangod/Zkd/ZkdHelper.cpp:356]: @maierlars is that better than `dim = dim % dimensions`?

Comment (maierlars) [reply arangod/Zkd/ZkdHelper.cpp:356]: perf says so. This is actually branch free using `cmovxx`.

Comment (goedderz) [reply arangod/Zkd/ZkdHelper.cpp:356]: Interesting, and unexpected.

**Comment by maierlars** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr/14508/

**Comment by maierlars** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr/14510/

File: arangod/Zkd/ZkdHelper.cpp [hunk 2]
```diff
@@ -240,61 +241,121 @@ auto zkd::compareWithBox(byte_string_view cur, byte_string_view min, byte_string
...
+-> bool {
+
+  if (dimensions == 0) {
+    auto msg = std::string{"dimensions argument to "};
+    msg += __func__;
+    msg += " must be greater than zero.";
+    throw std::invalid_argument{msg};
+  }
+
+  std::size_t max_size = std::max(std::max(cur.size(), min.size()), max.size());
```
Comment (goedderz) [inline arangod/Zkd/ZkdHelper.cpp:312]:
```suggestion
  std::size_t max_size = std::max({cur.size(), min.size(), max.size()});
```

**Review by goedderz — APPROVED** [review]

**Comment by maierlars** [comment]
Closed in favor of #13798

**Pull request closed without merging**